### PR TITLE
Fix(Kyverno): install CRDs before installing the Helm Release

### LINKED ISSUE #478: fix: Install Kyverno CRDs before installing Kyverno release
@tbobm they have split the crds in another chart that need to be install separately https://github.com/kyverno/kyverno/tree/main/charts

_Originally posted by @ArchiFleKs in https://github.com/particuleio/terraform-kubernetes-addons/issues/406#issuecomment-898488704_

### PR DESCRIPTION
**Pull request by tbobm**
Kyverno stores its CRDs in another Helm Chart

# Install Kyverno CRDs

## Description

Closes #478

Kyverno 2.0 separates the installation of Kyverno in the Kubernetes Cluster and the installation of the Custom Resource Definitions.
If required, install the CRDs beforehand.

### Checklist

- [x] CI tests are passing
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/particuleio/terraform-kubernetes-addons/#doc-generation

### CODE REVIEW

File: kyverno.tf [hunk 1]
```diff
@@ -58,6 +72,39 @@ resource "helm_release" "kyverno" {
   ]
   namespace = local.kyverno["create_ns"] ? kubernetes_namespace.kyverno.*.metadata.0.name[count.index] : local.kyverno["namespace"]
 
+  depends_on = [
+    helm_release.kube-prometheus-stack,
```
Comment (ArchiFleKs) [inline kyverno.tf:76]: I don't think dependencies are needed here

File: kyverno.tf [hunk 2]
```diff
@@ -58,6 +72,39 @@ resource "helm_release" "kyverno" {
...
+  skip_crds             = local.kyverno["skip_crds"]
+  verify                = local.kyverno["verify"]
+  values = [
+    local.values_kyverno,
+    local.kyverno["extra_values"]
+  ]
+  namespace = local.kyverno["create_ns"] ? kubernetes_namespace.kyverno.*.metadata.0.name[count.index] : local.kyverno["namespace"]
+
   depends_on = [
     helm_release.kube-prometheus-stack
```
Comment (ArchiFleKs) [inline kyverno.tf:62]: Should depends only on Prometheus CRD resources not the release. I need to fix it in all the other files. Dependencies are kind of a mess.

**Review by ArchiFleKs — CHANGES_REQUESTED** [review]
Nice :)

File: kyverno.tf (hunk 1 above)
Comment (tbobm) [reply kyverno.tf:76]: Well isn't the `helm_release.kyverno-crds` required here ?
I probably c/c'd the `helm_rease.kube-prometheus-stack` from another addon

File: kyverno.tf (hunk 2 above)
Comment (tbobm) [reply kyverno.tf:62]: I'm not sure I get why the Prometheus CRDs are required, is there a specific reason monitoring-related ?

**Comment by ArchiFleKs** [comment]
@tbobm I think for Kyverno you can remove all the depends for now. The logic behind the dependencies of some charts on kube-prometheus-stack or kube-prometheus-crd is because some chart can deploy a `ServiceMonitor` which is a custom resources from prometheus-operator, so if prometheus operator CRDs are not abailable chart install is failing.

At first CRD where installed directly with kube-prometheus-stack but now there are installed separately, so the charts that have a potential `ServiceMonitor` object will be dependent of prometheus-stack-crd.

For Kyverno I suggest you add the following : https://github.com/kyverno/kyverno/blob/main/charts/kyverno/values.yaml#L213

Here is a correct example of how it sould be done in : 
* https://github.com/particuleio/terraform-kubernetes-addons/blob/main/modules/aws/ingress-nginx.tf#L160
* https://github.com/particuleio/terraform-kubernetes-addons/blob/main/modules/aws/ingress-nginx.tf#L26

These features enable custom metrics collection out of the box for components that supports it when kube-prometheus is enable

**Review by ArchiFleKs — APPROVED** [review]

**Pull request merged**